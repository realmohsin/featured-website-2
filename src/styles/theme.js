import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
