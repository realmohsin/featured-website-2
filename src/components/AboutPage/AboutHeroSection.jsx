import React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import { GatsbyImage, getImage } from 'gatsby-plugin-image'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  example: {
    // ...mobile first styles,
    [theme.breakpoints.up('sm')]: {
      //...sm and up styles
    },
    [theme.breakpoints.up('md')]: {
      //...md and up styles
    },
    '@media (min-width: 1320px)': {
      //...rules for above 1320px
    }
  },
  aboutHeroSection: {
    marginTop: '-2rem',
    // position: 'relative',
    height: '42rem',
    overflow: 'hidden'
  },
  // AboutHeroSectionTitle: {
  //   zIndex: 1, // unclear why this is needed
  //   width: '100%',
  //   textAlign: 'center',
  //   position: 'absolute',
  //   top: '15%',
  //   color: theme.palette.primary.dark,
  //   fontFamily: ['Piazzolla', 'serif'].join(', '),
  //   fontSize: '4.2rem'
  // },
  heroImage: {
    transform: 'translateY(-10%)'
    // height: '100%',
    // objectFit: 'center',
    // objectPosition: '25% 25%'
  }
}))

const AboutHeroSection = () => {
  const classes = useStyles()
  const data = useStaticQuery(graphql`
    query {
      contentfulRomanoAboutPage {
        heroImage {
          gatsbyImageData(layout: FULL_WIDTH)
        }
      }
    }
  `)

  const image = getImage(data.contentfulRomanoAboutPage.heroImage)

  return (
    <section className={classes.aboutHeroSection}>
      <GatsbyImage
        image={image}
        layout='fullWidth'
        alt={'Romana Law Lawyers'}
        className={classes.heroImage}
      />
    </section>
  )
}

export default AboutHeroSection
