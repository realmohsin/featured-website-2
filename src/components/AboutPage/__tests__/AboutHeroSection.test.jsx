import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import AboutHeroSection from "../AboutHeroSection";

// Mock Gatsby's useStaticQuery and graphql
jest.mock("gatsby", () => ({
  useStaticQuery: jest.fn(() => ({
    contentfulRomanoAboutPage: {
      heroImage: {
        gatsbyImageData: {
          layout: "fullWidth",
          images: {
            fallback: {
              src: "/static/hero-image.jpg",
              srcSet: "/static/hero-image.jpg 1920w",
            },
          },
        },
      },
    },
  })),
  graphql: jest.fn(),
}));

// Mock gatsby-plugin-image
jest.mock("gatsby-plugin-image", () => ({
  GatsbyImage: ({ image, alt, className, layout, ...props }) => (
    <img
      src={image?.images?.fallback?.src || "/mock-image.jpg"}
      alt={alt}
      className={className}
      data-layout={layout}
      {...props}
    />
  ),
  getImage: jest.fn((image) => image),
}));

const theme = createMuiTheme({
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("AboutHeroSection Component", () => {
  it("renders hero section with image", () => {
    const { container } = renderWithTheme(<AboutHeroSection />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
