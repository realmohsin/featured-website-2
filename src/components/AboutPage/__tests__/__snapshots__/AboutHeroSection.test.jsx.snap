// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutHeroSection Component renders hero section with image 1`] = `
<section
  class="makeStyles-aboutHeroSection-1"
>
  <img
    alt="Romana Law Lawyers"
    class="makeStyles-heroImage-2"
    data-layout="fullWidth"
    src="/mock-image.jpg"
  />
</section>
`;
